### DISCUSSION TITLE
Effects to wamp up Single SID scenarios

### DISCUSSION THREAD

**Discussion by mwedmark** [frntc/SIDKick-pico#16] [Ideas]
I'm good at ideas, if you've noticed. After trying out different settings, both single and multi-SID ones, I got an idea. While playing/recording some songs, you could already setup 2 different SID playing the same thing and mix them in Stereo to your liking. Would it be possible to add some effect setting to get even a richer sound? Does the Pico still have some extra CC's to spare for something like that? F.e. Delay/Reverb/Chorus maybe..

**Comment by frntc** [comment]
I haven't thought about adding such effects. Simpler effects should be easy to implement. My (probably naive) understanding is that the compute time increases with the quality of these effects. Of course someone would have to add all this to the code and the config tools...

**Comment by magwed** [comment]
How much available RAM-memory can we spare to effects like delay/reverb? Of course these would only be needed if really enabled, like everything else in RAM. A Stereo-reverb effects would be out-of-this-world cool, but anything from the default effects family: Chorus/Flanger/Delay/Reverb will be a cool addition and simple ones comes almost for free.. but demands a bit of RAM..

I would probably start looking at something like this (code from an open-source Guitar multi-effects pedal written for Pi pico): 
https://github.com/StoneRose35/cortexguitarfx/tree/main/Src/pipicofx

**Comment by magwed** [comment]
Extra info here:
https://www.youtube.com/watch?v=jpfROA2EMzo&ab_channel=linsoft23

**Reply by frntc** [reply]
I think memory is not an issue (for most/simple effects), but the integration is a bit awkward (there's only one small audio buffer for DMA/DAC), and computation time is taken away from the SID-emulation making it less accurate.

**Comment by mwedmark** [comment]
Ahh ok, I though there were a bit of slack here. Thanks for the feedback! 

Another possible way of faking "effects" would be if it was possible to use the other SID in a single SID-scenario to fake a simple delay/reverb. This can already be done today of course, by doubling to SID-2 also at $D400 and mixing them a little different in stereo, spread them and lower one of them slightly. Something that would be cool is to be able to offset the second one in time in milliseconds and maybe put a default filter at a darker offset point. That would, kind of, mimic a reverb that was muffled by walls/furniture/carpets.. Well ideas, ideas!
